src/pe/delay_import.rs

```diff
@@ -0,0 +1,529 @@
+                offset,
+                dd.size,
+                bytes.len()
+            )));
+        }
+        let directory_bytes = &bytes[offset..offset + dd.size as usize];
+
+        let container = if is_64 {
+            Container::Big
+        } else {
+            Container::Little
+        };
+        // Avoid borrowing of sections here, we store the rest and modify sections field on demand
+        let pectx = PeCtx::new(container, scroll::LE, &[], file_alignment, *opts, bytes);
+
+        let ctx = Ctx::new(container, scroll::LE);
+
+        let mut cursor = 0;
+        let mut imports = Vec::new();
+        let mut descriptor =
+            directory_bytes.gread_with::<DelayImportDescriptor>(&mut cursor, scroll::LE)?;
+        loop {
+            if descriptor.is_null() || !descriptor.is_possibly_valid() {
+                break;
+            }
+
+            let dll_name_rva = descriptor.name_rva;
+            let dll_name_offset =
+                utils::find_offset(dll_name_rva as usize, sections, file_alignment, opts)
+                    .ok_or_else(|| {
+                        error::Error::Malformed(format!(
+                            "cannot map delay import dll name rva {dll_name_rva:#x}"
+                        ))
+                    })?;
+            let dll_name = bytes.pread::<&'a str>(dll_name_offset)?;
+
+            let name_table_rva = descriptor.name_table_rva;
+            let name_table_offset =
+                utils::find_offset(name_table_rva as usize, sections, file_alignment, opts)
+                    .ok_or_else(|| {
+                        error::Error::Malformed(format!(
+                            "cannot map delay import name table rva {name_table_rva:#x}"
+                        ))
+                    })?;
+
+            let mut index = 0;
+            let mut current_name_offset = name_table_offset;
+
+            'inner: loop {
+                if current_name_offset > bytes.len() {
+                    break 'inner;
+                }
+                let thunk = &bytes[current_name_offset..].pread_with::<DelayImportThunk>(0, ctx)?;
+                if thunk.is_null() {
+                    break 'inner;
+                }
+
+                let is_ordinal = if is_64 {
+                    thunk.is_ordinal64()
+                } else {
+                    thunk.is_ordinal32()
+                };
+                let ordinal = if is_ordinal { thunk.ordinal() } else { 0 };
+                let rva = descriptor.address_table_rva + (index * ctx.size()) as u32;
+
+                if is_ordinal {
+                    imports.push(DelayImportEntry {
+                        descriptor,
+                        offset: current_name_offset as u32,
+                        rva,
+                        hint: 0,
+                        dll: dll_name,
+                        name: None,
+                        ordinal,
+                    });
```

Comment:
i prefer the fields to be constructed/returned from ordinal, which is just:

```
ordinal, name, and hint
```

you can then have a single push, e.g.:

```rust
let (ordinal, name, hint) = if is_ordiinal {
  (thunk.ordinal(), None, 0)
} else {
 (0, // logic in other branch)
};
imports.push(DelayImportEntry {
     descriptor,
     offset: current_name_offset as u32,
     rva,
     hint,
     dll: dll_name,
     name,
     ordinal,,
 });
```